provider/datagen/src/transform/cldr/time_zones/names.rs

```diff
@@ -0,0 +1,196 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::convert::compute_bcp47_tzids_btreemap;
+use crate::transform::cldr::cldr_serde;
+use icu_provider::datagen::IterableDataProvider;
+use icu_provider::prelude::*;
+use icu_timezone::provider::names::*;
+use icu_timezone::TimeZoneBcp47Id;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::hash::Hasher;
+use zerotrie::ZeroTriePerfectHash;
+use zerovec::{ZeroSlice, ZeroVec};
+
+impl DataProvider<IanaToBcp47MapV1Marker> for crate::DatagenProvider {
+    fn load(&self, _: DataRequest) -> Result<DataResponse<IanaToBcp47MapV1Marker>, DataError> {
+        let resource: &cldr_serde::time_zones::bcp47_tzid::Resource =
+            self.cldr()?.bcp47().read_and_parse("timezone.json")?;
+
+        let iana2bcp = &compute_bcp47_tzids_btreemap(&resource.keyword.u.time_zones.values);
+
+        // Sort and deduplicate the BCP-47 IDs:
+        let bcp_set: BTreeSet<TimeZoneBcp47Id> = iana2bcp.values().copied().collect();
+        let bcp47_ids: ZeroVec<TimeZoneBcp47Id> = bcp_set.iter().copied().collect();
+        let bcp47_ids_checksum = compute_bcp47_ids_hash(&bcp47_ids);
+
+        // Transform the map to use BCP indices:
+        #[allow(clippy::unwrap_used)] // structures are derived from each other
+        let map: BTreeMap<Vec<u8>, usize> = iana2bcp
+            .iter()
+            .map(|(iana, bcp)| {
+                (
+                    iana.as_bytes().to_ascii_lowercase().to_vec(),
+                    bcp47_ids.binary_search(bcp).unwrap(),
+                )
+            })
+            .collect();
+
+        let data_struct = IanaToBcp47MapV1 {
+            map: ZeroTriePerfectHash::try_from(&map)
+                .map_err(|e| {
+                    DataError::custom("Could not create ZeroTrie from timezone.json data")
+                        .with_display_context(&e)
+                })?
+                .convert_store()
+                .into_zerotrie(),
+            bcp47_ids,
+            bcp47_ids_checksum,
+        };
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(data_struct)),
+        })
+    }
+}
+
+impl IterableDataProvider<IanaToBcp47MapV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(vec![Default::default()])
+    }
+}
+
+impl DataProvider<Bcp47ToIanaMapV1Marker> for crate::DatagenProvider {
+    fn load(&self, _: DataRequest) -> Result<DataResponse<Bcp47ToIanaMapV1Marker>, DataError> {
+        let resource: &cldr_serde::time_zones::bcp47_tzid::Resource =
+            self.cldr()?.bcp47().read_and_parse("timezone.json")?;
+        // Note: The BTreeMap retains the order of the aliases, which is important for establishing
```

Comment:
Note that the next CLDR release will include a new `iana` attribute that, if present, overrides the alias order. See https://github.com/unicode-org/cldr/pull/3105.

Does this PR handle that attribute?

Reply:
No, CLDR 44 Alpha is not in scope for the 1.3 release, but this will be a priority for the 1.4 release. Filed https://github.com/unicode-org/icu4x/issues/4044